sdcm/nemesis.py

```diff
@@ -4318,6 +4319,119 @@ def disrupt_add_remove_mv(self):
             self.unset_current_running_nemesis(cql_query_executor_node)
 
+    def disrupt_bootstrap_streaming_error(self):
+        new_node: BaseNode = self.cluster.add_nodes(
+            count=1, dc_idx=self.target_node.dc_idx, enable_auto_bootstrap=True, rack=self.target_node.rack)[0]
+        self.monitoring_set.reconfigure_scylla_monitoring()
+        self.set_current_running_nemesis(node=new_node)  # prevent to run nemesis on new node when running in parallel
+
+        def start_bootstrap(new_node: BaseNode, timeout=3600):
+            bootstrap_process = Process(target=self.cluster.node_setup,
+                                        name=f"Bootstraping_{new_node.name}",
+                                        kwargs={"node": new_node, "verbose": True, "timeout": int(timeout)},
+                                        daemon=True)
+            try:
+                self.log.info("Start node %s setup and bootstrapp", new_node.name)
+                bootstrap_log_follower = new_node.follow_system_log(patterns=['init - Startup failed',
+                                                                              'init - Startup interrupted',
+                                                                              'initialization completed'])
+                bootstrap_process.start()
+                log_messages = wait_for(func=lambda: list(bootstrap_log_follower),
+                                        text='Waiting for bootstap aborting or finishing', step=5, timeout=timeout)
+
+                if log_messages and ('init - Startup failed' in log_messages[0] or 'init - Startup interrupted' in log_messages[0]):
+                    self.log.debug("Bootstrap node %s was aborted", new_node.name)
+                    raise NodeSetupFailed(node=new_node, error_msg=log_messages[0])
+
+                if new_node.db_up() and new_node.jmx_up():
+                    self.log.info("Node %s bootstrapped succesfull", new_node.name)
+            except Exception as exc:  # pylint: disable=broad-except
+                self.log.warning("Node bootstrap was aborted with error: %s", exc)
+            finally:
+                self.log.debug("Stop node bootstrap process")
+                bootstrap_process.join(60)
+                bootstrap_process.terminate()
+                self.log.debug("Node bootstrap process stopped")
+
+        trigger = partial(
+            start_bootstrap, new_node=new_node)
+
+        terminate_patterns = self.target_node.raft.get_log_pattern_for_bootstrap_abort()
+
+        self.use_nemesis_seed()
+        terminate_pattern = random.choice(terminate_patterns)
+        self.log.info("Stop bootsrap process after log message: '%s'", terminate_pattern)
+
+        log_follower = new_node.follow_system_log(patterns=[terminate_pattern])
+
+        watcher = partial(
+            self._call_disrupt_func_after_expression_logged,
+            log_follower=log_follower,
+            disrupt_func=new_node.stop_scylla,
```

Comment:
Will we do only gracefully stop? what about the killing process or the stop node itself?

Reply:
It is possible to use both: reboot node(as for decommsision/rebuild streaming errors nemesises) or kill scylla process. In this case scylla just restarted and continue or if restart failed, the process will wait 1 hour in Parallel object runs. To avoid that stop scylla method was choosen. As additional enhancement, i can add to choose randomly how to interrupt scylla process